### DISCUSSION TITLE
[contribution] continuous collision check for the convex hull swept by the moving object in PyBullet

### DISCUSSION THREAD

**Discussion by mkarlan** [bulletphysics/bullet3#3749] [General]
I'd like to extend PyBullet with a useful new function that I need but have not managed to implement with the currently available functions.

Specifically, I'd like to calculate the closest point between a convex volume swept by a moving body A and a stationary body B. I need this as a collision detection query for motion planning. Attached you can find the envisioned interface of the new PyBullet function.

First of all, is it correct that such a function is not yet available in PyBullet?

Secondly, I have already started implementing this new PyBullet function using available functions from Bullet C++. Would you be interested in reviewing and merging a pull request? 

[pybullet getConvexSweepClosestPoint.docx](https://github.com/bulletphysics/bullet3/files/1748123/pybullet.getConvexSweepClosestPoint.docx)

**Comment by erwincoumans** [comment]
Thanks for bringing up the feature request.

Indeed, the swept collision query is not exposed yet. Your proposal in the docx looks reasonable. The work-flow to implement a new feature is not documented and non-trivial (Implement in PhysicsServerCommandProcessor.cpp, expose in Bullet-C-API and in pybullet). If you still want to give it a try, you are welcome and if the changes are limited I consider reviewing/merging. At some stage I may expose it myself too.
